### ISSUE TITLE
Fabric Support 

### ISSUE THREAD

**Issue by iamvucms** [react-native-linear-gradient/react-native-linear-gradient#552]
Will you have any plan to upgrade to support new  React Native fabric renderer system ?
[https://reactnative.dev/docs/next/new-architecture-intro](url)

**Comment by xrustic2020** [comment]
Now it`s really actually questions. New fabric renderer system now is available. We want migration to the new architectures, but we need react-native-linear-gradient. Please update support Fabric renderer system

**Comment by chakrihacker** [comment]
I started working on it, Hope I can provide some update if I am able to crack it

**Comment by chakrihacker** [comment]
Hey, small update iOS is working fine 🎉. 
I am facing some issues with building an Android example app that supports the fabric.

It might be related to these conversations

1. https://github.com/facebook/react-native/issues/33716
2. https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2370055

**Comment by capezzbr** [comment]
@chakrihacker keep us posted, 👏

**Comment by sebasg0** [comment]
Hi @chakrihacker if u need any help let me know

**Comment by chakrihacker** [comment]
@sebasg0 yeah I am actually stuck with android ndk issue when adding  the library to example 

Maybe we can connect and debug

**Comment by sebasg0** [comment]
sure lmk when available and we can take a look at

**Comment by chakrihacker** [comment]
Hey @sebasg0 i am available today and tomorrow let's connect

**Comment by sebasg0** [comment]
Sure reach me on discord bass#7266 @chakrihacker

**Comment by sebasg0** [comment]
Hi @chakrihacker, NDK issue solved. Lmk where to push my code so we can have this merged to master asap.

**Comment by sebasg0** [comment]
Hey @chakrihacker  https://github.com/FyndX/react-native-fabric-linear-gradient/pull/1 any other help needed now passed the idk issue lmk. I will be available tomorrow as Im on EST time.

**Comment by chakrihacker** [comment]
Hi 👋 

Here's the initial version 

Github: https://github.com/FyndX/react-native-gradient
npm: https://www.npmjs.com/package/@fyndx/react-native-gradient

I have plans to  improve it and add Radial Gradient, and web support in future

Please provide your feedback

**Comment by chakrihacker** [comment]
I don't have any plans yet

**Comment by capezzbr** [comment]
@chakrihacker the RN community will definitely appreciate if you could port your Fabric support over to this repo. That way we can keep using `react-native-linear-gradient` in the old/new architecture as we are migrating our codebases

**Comment by chakrihacker** [comment]
Hey @capezzbr I am personally aligned with breaking new and old architecture rather than supporting both at a time, I am ok to have a version bump that only supports new arch if time permits. 

But if you have any issues with fabric and Linear gradient i am happy to help

**Comment by capezzbr** [comment]
> I am personally aligned with breaking new and old architecture rather than supporting both at a time

@chakrihacker The issue I see with this approach is that it makes it hard testing new/old arch in medium/big RN applications, as that requires to bump dozen of dependencies instead of toggle a single flag. Which I believe is the reason why most native modules that today support the new arch are also retro-compatible.

**Comment by BraveEvidence** [comment]
I have got this working already check this https://www.youtube.com/watch?v=aohrzgRbrTQ&list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i&index=5 and https://github.com/BraveEvidence/ReactNativeNewArchLinearGradient2023/tree/main

**Comment by friederbluemle** [comment]
The **first prerelease version with new architecture support** has just been published: https://github.com/react-native-linear-gradient/react-native-linear-gradient/releases/tag/v3.0.0-alpha.0

I'm reopening this issue so people have a chance to test and report any issues. I will close it again once we have a stable version 3.0.0 published.

The prerelease is using the `next` dist tag on npm, so a regular `npm i ..`/`yarn add ..` will still only give you the latest stable build.

In order to **test this new prerelease**, use the following commands:

**Yarn**

```
yarn add react-native-linear-gradient@next
```

**npm**

```sh
npm i react-native-linear-gradient@next
```

Please reply here if something is not working. Thanks!

**Comment by jackstudd** [comment]
@friederbluemle everything working fine here on `3.0.0-alpha.1` with the following simple use case:

```
<LinearGradient
  style={{
    position: 'absolute',
    height: '100%',
    width: '100%',
  }}
  colors={['transparent', Colors.background]}
/>
```
Thanks a lot !

**Comment by JeanSilvany** [comment]
> The **first prerelease version with new architecture support** has just been published: https://github.com/react-native-linear-gradient/react-native-linear-gradient/releases/tag/v3.0.0-alpha.0
> 
> I'm reopening this issue so people have a chance to test and report any issues. I will close it again once we have a stable version 3.0.0 published.
> 
> The prerelease is using the `next` dist tag on npm, so a regular `npm i ..`/`yarn add ..` will still only give you the latest stable build.
> 
> In order to **test this new prerelease**, use the following commands:
> 
> **Yarn**
> 
> ```
> yarn add react-native-linear-gradient@next
> ```
> 
> **npm**
> 
> ```shell
> npm i react-native-linear-gradient@next
> ```
> 
> Please reply here if something is not working. Thanks!

Thanks, works fine!

**Comment by aladine** [comment]
When the official version 3.0.0 is published? I saw 3.0.0-alpha.1 is published on Aug 2023 and we are already in 2025
